src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,306 @@
+import React, {Component, PropTypes} from 'react';
+import { DateRangePicker } from 'react-dates';
+import moment from 'moment';
+
+/**
+ * DatePickerRange is a tailor made component designed for selecting
+ * timespan across multiple days off of a calendar.
+ *
+ * The DatePicker integrates well with the Python datetime module with the
+ * startDate and endDate being returned in a string format suitable for
+ * creating datetime objects.
+ *
+ * This component is based off of Airbnb's react-dates react component
+ * which can be found here: https://github.com/airbnb/react-dates
+ */
+export default class DatePickerRange extends Component {
+    constructor(props) {
+      super(props);
+      this.props.startDate = this.props.start_date;
+      this.props.endDate = this.props.end_date;
+      this.props.initialVisibleMonth = this.props.initial_visible_month;
+      this.props.minDateRange = this.props.min_date_range;
+      this.props.maxDateRange = this.props.max_date_range;
```

Comment:
we should treat `props` as immutable - never modifying elements or creating new ones. Otherwise, the parent component (who is passing in `props` to this component) might end up with `props` in a different structure and that can cause bugs.

So we should remove all of these `this.props.` assignements